File: java/src/main/java/P063.java
import java.math.BigInteger;
import java.util.HashMap;
import java.util.Map;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;

@SuppressWarnings("unused")
public class P063 implements Puzzle {
    @Override
    public String solve() {
        return "" + IntStream.range(1, 50)
                .parallel()
                .map(i -> count(i))
                .sum();
    }

    private static int count(int exponent) {
        int count = 0;

        final BigInteger min = BigInteger.TEN.pow(exponent - 1);
        final BigInteger max = BigInteger.TEN.pow(exponent);

        for (BigInteger base = BigInteger.ZERO; ; base = base.add(BigInteger.ONE)) {
            BigInteger result = base.pow(exponent);
            if (result.compareTo(min) >= 0) {
                if (result.compareTo(max) < 0) {
                    ++count;
                } else {
                    break;
                }
            }
        }
        return count;
    }

}

File: python/p063.py
def g(power):
    count = 0
    i = 1
    min = 10**(power - 1)
    max = 10**power - 1

    while True:
        result = i**power
        if result >= min:
            if result <= max:
                count += 1
            else:
                break
        i += 1

    return count

count = 0
for i in xrange(1, 1000):
    current = g(i)
    if current > 0:
        count += current
    else:
        break

print count

